_docs/technical/protocols/2017-01-01-application-level.md

```diff
@@ -110,14 +110,18 @@ module for the examples of messages that are using `Inv/Req/Data`.
 
 ## Block Exchange Messages
 
-This table explains [Pos.Block.Network.Types](https://github.com/input-output-hk/cardano-sl/blob/d564b3f5a7e03e086b62c88212870b5ea89f5e8b/src/Pos/Block/Network/Types.hs) module.
+[//]: # (Updated at 3b657302dede832b908f7ba792a164c83b362712)
 
-| Message type | Payload | Commentaries |
-|--------------|---------|--------------|
-| GetHeaders | Oldest header hash we're interested in; Newest hash we're interested in, or blank | Expect newest header first |
-| GetBlocks | Oldest header hash; Newest hash | As opposed to GetHeaders, both hashes have to be present |
-| BlockHeaders | Non-empty collection of block headers, newest first | Polymorphic in scc |
-| Block | A single block | Polymorphic in scc |
+This table explains [Pos.Block.Network.Types](https://github.com/input-output-hk/cardano-sl/blob/3b657302dede832b908f7ba792a164c83b362712/src/Pos/Block/Network/Types.hs) module.
+
+| Message type | Payload                                                           | Commentaries                                             |
+|--------------+-------------------------------------------------------------------+----------------------------------------------------------|
+| GetHeaders   | Header hash checkpoints; Optional newest hash we're interested in | Expect newest header first                               |
+| GetBlocks    | Oldest header hash; Newest hash                                   | As opposed to GetHeaders, both hashes have to be present |
```

Comment:
I don't like this _As opposed to GetHeaders_. These messages have different arguments and different semantics. I would understand it if `GetHeaders` had two optional hashes (or one optional and one mandatory). But `GetHeaders` has a list of checkpoints and optional newest hash, while `GetBlocks` has two hashes which specify range.